components/builder-core/src/metrics.rs

```diff
@@ -138,117 +144,3 @@ fn statsd_client() -> Option<Client> {
         Err(_) => None,
     }
 }
-
-impl Counter {
-    pub fn increment(&self) {
-        match sender().send((
-            MetricType::Counter,
-            MetricOperation::Increment,
-            &self.id(),
-            None,
-        )) {
-            Ok(_) => (),
-            Err(e) => error!("Failed to increment counter, error: {:?}", e),
-        }
-    }
-
-    pub fn decrement(&self) {
-        match sender().send((
-            MetricType::Counter,
-            MetricOperation::Decrement,
-            &self.id(),
-            None,
-        )) {
-            Ok(_) => (),
-            Err(e) => error!("Failed to decrement counter, error: {:?}", e),
-        }
-    }
-}
-
-impl Gauge {
-    pub fn set(&self, val: f64) {
-        match sender().send((
-            MetricType::Gauge,
-            MetricOperation::SetValue,
-            &self.id(),
-            Some(val),
-        )) {
-            Ok(_) => (),
-            Err(e) => error!("Failed to set gauge, error: {:?}", e),
-        }
-    }
-}
-
-impl Metric for Counter {
-    fn id(&self) -> &'static str {
-        match *self {
-            Counter::GithubApi => "github.api",
-            Counter::GithubAuthenticate => "github.authenticate",
-            Counter::GithubEvent => "github.event",
-            Counter::GithubInstallationToken => "github.installation-token",
-            Counter::SearchPackages => "search-packages",
-        }
-    }
-}
-
-impl Metric for Gauge {
-    fn id(&self) -> &'static str {
-        match *self {
-            Gauge::PackageCount => "package-count",
-        }
-    }
-}
-
-#[cfg(test)]
```

Comment:
Tests would be nice to preserve as well..

Reply:
None of these tests actually did anything meaningful, and most were ignored. I could try to add meaningful ones, but it may take a while, and would likely end up just testing the statsd client we're using.

I suppose implementing things like `incr` in the `CounterMetric` trait could be useful for that, since we'd be able to more easily test the behavioral contract.

Reply:
Yeah they were the barest minimum I could think of.. we can add some in the next pass.